### PR TITLE
[main] Aligning `openssl` versions across toolchain build phases.

### PR DESCRIPTION
**Pull request by PawelWMS**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

While working on #2515, @mfrw noticed that we're building the 1.1.1g version of `openssl` in the 1<sup>st</sup> phase of the toolchain builds, while in general Mariner is on the 1.1.1k version. This PR makes the built versions consistent.

###### Change Log  

- Made sure we build the 1.1.1k version of `openssl` consistently across toolchain build phases.

###### Does this affect the toolchain?  

Yes.

###### Associated issues  

- #2515

###### Test Methodology

- Pipeline toolchain build: 175989

### CODE REVIEW

File: toolkit/scripts/toolchain/container/toolchain-sha256sums [hunk 1]
```diff
@@ -42,7 +42,7 @@ c2faa85e625cd7567dc1dec9f23d1af1f73e2ca6468bce1bdc13387ec56f97e8  man-pages-5.02
 0c98a3f1732ff6ca4ea690552079da9c597872d30e96ec28414ee23c95558a7f  mpfr-4.1.0.tar.xz
 30306e0c76e0f9f1f0de987cf1c82a5c21e1ce6568b9227f7da5b71cbea86c9d  ncurses-6.2.tar.gz
 15ea32c7b100217b6e3193bc03e77f485d9bf7504051443ba9ce86d1c17c6b5a  nspr-4.21.tar.gz
-ddb04774f1e32f0c49751e21b67216ac87852ceb056b75209af2443400636d46  openssl-1.1.1g.tar.gz
+892a0875b9872acd04a9fde79b1f943075d5ea162415de3047c327df33fbaee5  openssl-1.1.1k.tar.gz
```
Comment (thcrain-msft) [inline toolkit/scripts/toolchain/container/toolchain-sha256sums:45]: Can you make sure you add the new source tarball into the `toolkit` folder in the blob store?

**Review by thcrain-msft — APPROVED** [review]
LGTM, just one request!

File: toolkit/scripts/toolchain/container/toolchain-sha256sums (hunk 1 above)
Comment (PawelWMS) [reply toolkit/scripts/toolchain/container/toolchain-sha256sums:45]: Ah, right, thanks, done.:)

Comment (thcrain-msft) [reply toolkit/scripts/toolchain/container/toolchain-sha256sums:45]: Thanks! I ran into a couple of signature match errors due to URL weirdness on a sourceforge download last night, so I went and uploaded most of the missing ones today. Just wanna make sure we keep it up-to-date!

**Review by mfrw — APPROVED** [review]
LGTM :)

**Pull request merged**